src/layouts/Folders.jsx

```diff
@@ -56,28 +69,65 @@ const Folders = ({ match, location }) => {
           );
         }
       }
-    }, [error])
+    }, [queryError])
+
+    useEffect(() => {
+      if (mutateError) {
+        toast(
+          <Toast notificationType='error' text={`Your file reordering could not be saved. Please try again. ${DEFAULT_ERROR_TOAST_MSG}`}/>,
+          {className: `${elementStyles.toastError} ${elementStyles.toastLong}`},
+        );
+      }
+    }, [mutateError])
 
     useEffect(() => {
         if (folderContents && folderContents.data) {
           const parsedFolderContents = parseDirectoryFile(folderContents.data.content)
           setDirectoryFileSha(folderContents.data.sha)
+          setParsedFolderContents(parsedFolderContents)
 
           if (subfolderName) {
             const subfolderFiles = retrieveSubfolderContents(parsedFolderContents, subfolderName)
             if (subfolderFiles.length > 0) {
-              setFolderOrder(retrieveSubfolderContents(parsedFolderContents, subfolderName))
+              setFolderOrderArray(subfolderFiles)
             } else {
               // if subfolderName prop does not match directory file, it's not a valid subfolder
               if (!shouldRedirect) setShouldRedirect(true)
             }
           } else {
-            setFolderOrder(convertFolderOrderToArray(parsedFolderContents))
+            setFolderOrderArray(convertFolderOrderToArray(parsedFolderContents))
           }
         }
     }, [folderContents, subfolderName])
 
-    const toggleRearrange = () => { setIsRearrangeActive((prevState) => !prevState) }
+    const toggleRearrange = () => { 
+      // if no folder contents, do not enable reordering
+      if (folderOrderArray.length === 0 || !folderContents) return
+
```

Comment:
As discussed, we might want to move this logic into conditional disabling of the button instead, since currently the button appears clickable even when reordering is disabled

Reply:
fixed in 4987834